fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/NormalizedLuceneSearchProvider.java

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension.impl;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRow;
+import com.yahoo.bard.webservice.util.DimensionStoreKeyUtils;
+import com.yahoo.bard.webservice.util.Pagination;
+import com.yahoo.bard.webservice.web.util.PaginationParameters;
+import com.yahoo.bard.webservice.data.dimension.SearchQuerySearchProvider;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.miscellaneous.PerFieldAnalyzerWrapper;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.queryparser.simple.SimpleQueryParser;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Lucene search provider that also supports text search queries.
+ */
+public class NormalizedLuceneSearchProvider extends LuceneSearchProvider implements SearchQuerySearchProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(NormalizedLuceneSearchProvider.class);
+
+    public static final String SEARCH_COLUMN_NAME = "__search";
+    public static final Analyzer DIACRITIC_ANALYZER = new DiacriticNormalizingAnalyzer();
+
+    private SimpleQueryParser queryParser;
+
+    private boolean searchColumnExists;
+    private IndexSearcher lastIndexSearcher;
+
+    /**
+     * Constructor.
+     *
+     * @param luceneIndexPath  Path to the lucene index files
+     * @param maxResults  Maximum number of allowed results in a page
+     * @param searchTimeout  Maximum time in milliseconds that a lucene search can run
+     */
+    public NormalizedLuceneSearchProvider(String luceneIndexPath, int maxResults, int searchTimeout) {
+        super(luceneIndexPath, maxResults, searchTimeout);
+
+        // override analyzer in LuceneSearchProvider
+        Map<String, Analyzer> analyzerMap = new HashMap<>();
+        analyzerMap.put(SEARCH_COLUMN_NAME, DIACRITIC_ANALYZER);
+        analyzer = new PerFieldAnalyzerWrapper(STANDARD_LUCENE_ANALYZER, analyzerMap);
```

Comment:
Rather than setting the field directly, I think it would be *much* clearer if we either gave
`LuceneSearchProvider` a protected `setAnalyzer` method, or even better create a protected constructor
that can be used by subclasses to override the analyzer.
This class can have static helper function to build its analyzer (since it
requires multiple statements). Modifying a super class' fields willy-nilly is a
recipe for hard-to-understand code.